src/iceberg/catalog/rest/config.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "cpr/cpr.h"
+#include "iceberg/catalog/rest/iceberg_rest_export.h"
+#include "iceberg/catalog/rest/util.h"
+#include "iceberg/result.h"
+#include "iceberg/table_identifier.h"
+
+/// \file iceberg/catalog/rest/config.h
+/// \brief RestCatalogConfig implementation for Iceberg REST API.
+
+namespace iceberg::rest {
+
+/// \brief Configuration struct for a Rest Catalog.
+///
+/// This struct holds all the necessary configuration for connecting and interacting with
+/// a Rest service. It's a simple data structure with public members that can be
+/// initialized directly via aggregate initialization. It provides helper methods to
+/// construct specific endpoint URLs and HTTP headers from the configuration properties.
+struct ICEBERG_REST_EXPORT RestCatalogConfig {
```

Comment:
```suggestion
struct ICEBERG_REST_EXPORT RestCatalogConfig : public ConfigBase<RestCatalogConfig> {
```

Consider extending `ConfigBase` for all key-value properties. This will be a lot easier to add predefined property like https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/rest/RESTCatalogProperties.java and https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/CatalogProperties.java.

See `src/iceberg/table_properties.h` on how table properties are defined.